### ISSUE TITLE
NRF Based M3 Tag AP firmware?

### ISSUE THREAD

**Issue by 20cstrothman** [jjwbruijn/OpenEPaperLink#279]
I recently got a Solum EL029H4WRC tag (N52811 chip) and was able to flash the SOL_M3_Uni_full_0027 firmware on it. It appears to work, but I do not have a tag that will work with any of the AP firmware. I was wondering if there are any plans for (or if there is any) AP firmware for the NRF based M3 tags? 
![20240325_161315](https://github.com/jjwbruijn/OpenEPaperLink/assets/29297335/d28f62ed-60fd-45b9-93eb-e04ac1fee84e)
![20240325_161326](https://github.com/jjwbruijn/OpenEPaperLink/assets/29297335/37d06fbb-ef5a-40b0-8c87-19bcacdeaafe)
![20240325_161348](https://github.com/jjwbruijn/OpenEPaperLink/assets/29297335/4a21470e-2896-47a4-a1b7-46e4006935cc)
![20240325_161513](https://github.com/jjwbruijn/OpenEPaperLink/assets/29297335/50c1efc0-c7cb-43ba-a924-486b06baa405)

**Comment by nlimper** [comment]
If you have a working OEPL access point within range, the tag will see it. No special firmware needed.
I don't know about this type, but some tag types have the zigbee antenna in the case. If you remove the pcb from the case, you don't have an antenna. If the antenna situation is solved, take a look at your AP to see if it's working at all.

**Comment by atc1441** [comment]
Here is the AP Firmware for the M3 hardware: https://github.com/jjwbruijn/OpenEPaperLink/tree/master/ARM_Tag_FW/nrf52811_Platformio_AP

It will only work as an AP and will not drive the display connecred to the AP PCB

**Comment by 20cstrothman** [comment]
Thanks for the info. @atc1441 quick question, what all do I need to do to get the bin for the AP firmware?

**Comment by atc1441** [comment]
PlatformIO

**Comment by 20cstrothman** [comment]
Alright, this is where I've gotten. Might you all know what I should do from here? It won't build because of this...
![image](https://github.com/jjwbruijn/OpenEPaperLink/assets/29297335/a4bf6a7c-26e0-45ad-ab14-37ae6fbc7c28)

**Comment by 20cstrothman** [comment]
As an update for this, I didn't realize there were different pinouts for the flasher and the access point (duh). Once I got my computer settled and the pins connected the right way, everything worked smoothly (at least with these tags). I have some 4.2" tags, but they're not responding to the flasher for reading or writing. Also, the errors I was getting before were because I was trying to compile it on a Windows machine. I whipped up a Ubuntu VM and compiled it in there with no issues. 

Here is the pinout for the flasher
![20240408_010910](https://github.com/jjwbruijn/OpenEPaperLink/assets/29297335/3e374bb0-74b1-4852-b55d-403d70b1b784)

Here is the pinout for the AP
![20240408_010954](https://github.com/jjwbruijn/OpenEPaperLink/assets/29297335/dfbc629a-6822-4858-b1fb-db9cfbc00a8b)
